### PR TITLE
Initial draft for web-optimized walkthrough

### PR DESCRIPTION
**Pull request by digitarald**
Ensuring the items shown on web work and are contextual.

cc @bamurtaugh

### CODE REVIEW

**Comment by bpasero** [comment]
@digitarald any thoughts on bringing back "Open File" to welcome? It will work in web, even when no remote is opened.

File: src/vs/workbench/contrib/welcome/gettingStarted/common/gettingStartedContent.ts [hunk 1]
```diff
@@ -178,11 +188,20 @@ export const walkthroughs: GettingStartedWalkthroughContent = [
 						type: 'svg', altText: 'Workspace Trust editor in Restricted mode and a primary button for switching to Trusted mode.', path: 'workspaceTrust.svg'
 					},
 				},
+				{
+					id: 'pickAFolderTask-Web',
+					title: localize('gettingStarted.setup.OpenFolder.title', "Open up your code"),
+					description: localize('gettingStarted.setup.OpenFolderWeb.description.interpolated', "You're all set to start coding. Open a project folder to get your files into VS Code.\n{0}\n{1}", Button(localize('openLocalFolder', "Open Local Folder"), 'command:workbench.action.addRootFolder'), Button(localize('openRepository', "Open Repository"), 'command:remoteHub.openRepository')),
+					// when: 'isWeb && workspaceFolderCount == 0',
```
Comment (gjsjohnmurray) [inline src/vs/workbench/contrib/welcome/gettingStarted/common/gettingStartedContent.ts:195]: why commented out?

**Comment by bpasero** [comment]
Also, "Add Folder" will work in serverless unless connected to GitHub.

File: src/vs/workbench/contrib/welcome/gettingStarted/common/gettingStartedContent.ts (hunk 1 above)
Comment (digitarald) [reply src/vs/workbench/contrib/welcome/gettingStarted/common/gettingStartedContent.ts:195]: My bad, this was just to make it visible during testing and why I left the pull request in review.

File: src/vs/workbench/contrib/welcome/gettingStarted/common/gettingStartedContent.ts [hunk 2]
```diff
@@ -178,11 +197,20 @@ export const walkthroughs: GettingStartedWalkthroughContent = [
 						type: 'svg', altText: 'Workspace Trust editor in Restricted mode and a primary button for switching to Trusted mode.', path: 'workspaceTrust.svg'
 					},
 				},
+				{
+					id: 'pickAFolderTask-Web',
+					title: localize('gettingStarted.setup.OpenFolder.title', "Open up your code"),
+					description: localize('gettingStarted.setup.OpenFolderWeb.description.interpolated', "You're all set to start coding. Open a project folder to get your files into VS Code.\n{0}\n \n{1}", Button(localize('openLocalFolder', "Open Local Folder"), 'command:workbench.action.addRootFolder'), Button(localize('openRepository', "Open Repository"), 'command:remoteHub.openRepository')),
```
Comment (digitarald) [inline src/vs/workbench/contrib/welcome/gettingStarted/common/gettingStartedContent.ts:203]: @JacksonKearl I had to add the space in `\n \n` to make the buttons not touch each other. Is that a viable solution or is there something else?

Comment (JacksonKearl) [reply src/vs/workbench/contrib/welcome/gettingStarted/common/gettingStartedContent.ts:203]: Pushed up CSS to resolve this.

**Comment by bamurtaugh** [comment]
Thanks so much for creating and merging, @digitarald! Looks great to me.

Small thought - the extensions section shows popular extensions, but the button says to browse recommended:

`Button(localize('browseRecommended', "Browse Recommended Extensions")`

However, I could also see that we're recommending users to take a look at popular ones, so saying `Popular` or `Recommended` either way feels accurate?

**Comment by digitarald** [comment]
> However, I could also see that we're recommending users to take a look at popular ones, so saying Popular or Recommended either way feels accurate?

Yes, this is why I left the copy in the step as is. Popular/Recommended shouldn't make much of a difference to users and is more of an implementation detail. If we get feedback on it and get a more curated web extension list, we can change it – especially if we get a better idea of extensions from the blog post.

As I write this, it might be worth pointing out, for example, that only a subset of extensions work. I do another pull request for that one.

**Pull request merged**